HHS/OPRE-OPS
# Ops 1462 refactor tokens frontend

## What changed

* Created `getAccessToken()` and `getRefreshToken()` as wrappers for accessing the tokens; future updates may change where the tokens live, so this makes it easier to refactor that later.
* Created checks, on the frontend, so that if the `access_token` is expired, it will actually call the backend's `/refresh` endpoint, and get a new access token
* Added `iss` and `aud` claims to our tokens (via default configs in `flask_jwt_extended` config
* Added an `iss` check, with temporary values, simply to help fill out the logic
* Created a `ValidateTokenStatus` object, to capture the current, and future status of tokens when checked.
* Created unit-tests for the `isValidToken` function
* Updated the `callBackend()` function, now takes an optional parameter, which determines if the `access_token` is used or `refresh_token`; defaults to `false` which is `access_token`; :information: I thought about simply passing in the functions themselves, it would be more scalable, but in reality we only have two token types.

## Issue

#1462

## How to test

1. Login with your preferred provider (`FakeAuth`)
2. Wait the desired time that the access_token should live (default is 30 min). 
3. After that time period, you should still have a valid session, but a new access_token.
4. The access_token["fresh"] value should be `false`


## Definition of Done Checklist
